src/compiler.js

```diff
@@ -103,14 +104,17 @@ module.exports = class GherkinTestcafeCompiler {
     this.afterAllHooks = finalizedStepDefinitions.afterTestRunHookDefinitions;
     this.beforeHooks = finalizedStepDefinitions.beforeTestCaseHookDefinitions;
     this.beforeAllHooks = finalizedStepDefinitions.beforeTestRunHookDefinitions;
-
     this.stepDefinitions = finalizedStepDefinitions.stepDefinitions;
   }
 
   _resolveAndRunStepDefinition(testController, step) {
     for (const stepDefinition of this.stepDefinitions) {
       const [isMatched, parameters] = this._shouldRunStep(stepDefinition, step);
       if (isMatched) {
+        if (this._hasDatatable(step)){
+          const table = new dataTable.default(step.arguments[0]);
+          return this._runStep(stepDefinition.code, testController, table);
```

Comment:
Please add the data table to `_shouldRunStep` and always pass it as a third parameter (`null` if there is no data table).

Reply:
implemented in _shouldRunStep

Reply:
Please make it the third parameter, so that one can use regex parameters and tables at the same time:

```suggestion
if (isMatched) {
  return this._runStep(stepDefinition.code, testController, parameters, table);
}
```

Reply:
ok I got what you meant now good suggestion